ckan/lib/helpers.py

```diff
@@ -1126,8 +1127,55 @@ def _make_menu_item(menu_item, title, **kw):
 
 
 @core_helper
-def default_group_type():
-    return str(config.get('ckan.default.group_type', 'group'))
+def default_group_type(type_='group'):
+    """Get default group/organization type for using site-wide.
+    """
+    return str(config.get('ckan.default.{}_type'.format(type_), type_))
+
+
+@core_helper
+def humanize_entity_type(entity_type, object_type, purpose):
+    """Convert machine-readable representation of package/group type into
+    human-readable form.
+    Example::
+
+      >>> humanize_entity_type('group', 'custom_group', 'add link')
+      'Add Custom Group'
+      >>> humanize_entity_type('group', 'custom_group', 'breadcrumb')
+      'Custom Groups'
+
+    """
+    if entity_type == object_type:
+        return  # use the default text included in template
+
+    log.debug(
+        u'Humanize %s of type %s for %s', entity_type, object_type, purpose)
+    templates = {
+        u'add link': _(u"Add {object_type}"),
+        u'breadcrumb': _(u"{object_type}s"),
+        u'facet label': _(u"{object_type}s"),
+        u'page title': _(u"{object_type}s"),
+        u'main nav': _(u"{object_type}s"),
+        u'create title': _(u"Create {object_type}"),
+        u'content tab': _(u"{object_type}s"),
+        u'default label': _(u"{object_type}"),
+        u'no label': _(u"No {object_type}"),
+        u'form label': _(u"{object_type} Form"),
+        u'edit label': _(u"Edit {object_type}"),
+        u'update label': _(u"Update {object_type}"),
+        u'create label': _(u"Create {object_type}"),
+        u'save label': _(u"Save {object_type}"),
+        u'sidebar label': _(u"{object_type}"),
+        u'my label': _(u"My {object_type}s"),
+        u'no any objects': _(
+            u"There are currently no {object_type}s for this site"),
+        u'description placeholder': _(
+            u"A little information about my {object_type}..."),
+    }
+    if purpose not in templates:
+        return
```

Comment:
do you want to use this default path for the no-op humanize things like "sidebar label" or "default label" so they don't show up in the dict above with somewhat meaningless `_("{object_type}")` "translations". Also if our default just shows the object type then a new purpose won't return `None`

Reply:
Sorry, didn't get it

This is sidebar label - 
![image](https://user-images.githubusercontent.com/11091199/86820871-4eeaf280-c092-11ea-98d8-b815a24163c9.png)

When dataset belongs to the organization with type `publisher` it has the sense to show `Publisher` as a sidebar label and that's what happens now. It's not a problem to remove those keys from dict and return `None`. In that case, the sidebar label will be `Organization`(default from template) but as long as we can redefine helper, it can be fixed.

Returning object type instead `None` as fallback won't work. There are a number of long messages, that are not added to the dict ([delete confirmation](https://github.com/ckan/ckan/pull/5337/files#diff-ea325fbc1a770c95b6cffacc7f23ecb7R39) ) and we have to return `None` in order to use template's default and not put such rare but big messages into dict.

Does it correlates somehow with your question?

Reply:
of course, you're right I forgot about the default outside the caller.